ui/src/kapacitor/components/config/SlackConfig.tsx

```diff
@@ -183,29 +214,31 @@ class SlackConfig extends PureComponent<Props, State> {
   }
 
   private handleSubmit = async e => {
-    const {isNewConfig} = this.props
     e.preventDefault()
+
+    const {workspace} = this.state
+    const {isNewConfig} = this.props
+
     const properties: SlackProperties = {
       url: this.url.value,
       channel: this.channel.value,
       enabled: this.state.enabled,
     }
+
     if (isNewConfig) {
-      properties.workspace = this.workspace.value
+      properties.workspace = workspace
```

Comment:
1. a bit strange seeing a value on an object being assigned directly -- could a spread operator be used here?
1. also in the context of all of these getters, this reads a bit weird to me -- what do you think about `workspace` --> `this.workspace` and putting workspace behind a getter?

Reply:
we only want to send the workspace on create, for updates we want to omit the key altogether.